src/main/java/edu/byu/cs/autograder/score/Scorer.java

```diff
@@ -286,13 +286,31 @@ private boolean passed(Rubric rubric) {
         return true;
     }
 
-    private Rubric applyLatePenalty(Rubric rubric, int daysLate) {
+    private Rubric applyLatePenalty(Rubric rubric, int daysLate) throws DataAccessException {
+        Submission previousSubmission = DaoService.getSubmissionDao().getBestSubmissionForPhase(gradingContext.netId(), gradingContext.phase());
         EnumMap<Rubric.RubricType, Rubric.RubricItem> items = new EnumMap<>(Rubric.RubricType.class);
         float lateAdjustment = daysLate * PER_DAY_LATE_PENALTY;
         for(Map.Entry<Rubric.RubricType, Rubric.RubricItem> entry : rubric.items().entrySet()) {
             Rubric.Results results = entry.getValue().results();
-            results = new Rubric.Results(results.notes(),
-                    results.score() * (1 - lateAdjustment),
+
+            String notes = results.notes();
+            float score = results.score() * (1 - lateAdjustment);
+
+            if(previousSubmission != null) {
+                Rubric.RubricItem previousItem = previousSubmission.rubric().items().get(entry.getKey());
+                if(previousItem != null &&
+                        previousItem.results().score() > score &&
+                        previousItem.results().rawScore() <= results.rawScore()) {
+                    notes = String.format("%s\nDeferring to less-penalized prior score of %s/%d",
+                            previousItem.results().notes().trim(),
+                            Math.round(previousItem.results().score() * 100) / 100.0,
+                            previousItem.results().possiblePoints());
```

Comment:
At first, I was surprised to see the double being formatted as a `%s`. After playing around with it for awhile, I discovered that it actually is the best approach because of the way it handles variable numbers of decimal places.

Here are the results of my researching:

```java
double result = 9.00;
System.out.println(String.format("Result: %f", result));
System.out.println(String.format("Result: %2.1f", result));
System.out.println(String.format("Result: %2.2f", result));
System.out.println(String.format("Result: %f", Math.round(result*100) / 100.0));
System.out.println(String.format("Result: %s", Math.round(result*100) / 100.0));

// #####################

// Result: 9.000000
// Result: 9.0
// Result: 9.00
// Result: 9.000000
// Result: 9.0
```

Reply:
Yep. I did the same thing. Maybe I should have left a comment.